course_info/static/course_info/js/controllers/PeopleController.js

```diff
@@ -466,60 +556,55 @@
-                courseInstance['term'] = ci.term ? ci.term.display_name : '';
-                courseInstance['year'] = ci.term ? ci.term.academic_year : '';
-                courseInstance['course_instance_id'] = ci.course_instance_id;
-                courseInstance['registrar_code_display'] = ci.course ?
-                        (ci.course.registrar_code_display
-                        || ci.course.registrar_code).trim() : '';
-                if (ci.secondary_xlist_instances &&
-                    ci.secondary_xlist_instances.length > 0) {
-                        courseInstance['xlist_status'] = 'Primary';
-                } else if (ci.primary_xlist_instances &&
-                    ci.primary_xlist_instances.length > 0) {
-                        courseInstance['xlist_status'] = 'Secondary';
-                } else {
-                        courseInstance['xlist_status'] = 'N/A';
-                }
-                var sites = ci.sites || [];
-                var siteIds =[];
-                sites.forEach(function (site) {
-                    site.site_id = site.external_id;
-                    if (site.site_id.indexOf('http') === 0) {
-                        site.site_id = site.site_id.substr(site.site_id.lastIndexOf('/')+1);
-                    }
-                    siteIds.push(site.site_id)
-                });
-                courseInstance['sites']= siteIds.length>0 ? siteIds.join(', ') : 'N/A';
+        $scope.showAddNewMemberResults = function() {
+            /* Updates page with summary message and failure details after all
+             add person operations are finished.
+             */
+            // use 'total_failures' to avoid stomping existing 'failures'
+            $scope.tracking.total_failures = $scope.tracking.total -
+                                             $scope.tracking.successes;
+            // figure out the alert type (ie. the color) here
+            var alertType = '';
+            if ($scope.tracking.successes == 0) {
+                alertType = 'danger';
             }
-            return courseInstance;
+            else if ($scope.tracking.total_failures == 0) {
+                alertType = 'success';
+            }
+            else {
+                alertType = 'warning';
+            }
+            $scope.messages.success = {type: 'add', alertType: alertType};
+            if ($scope.tracking.successes) { $scope.dtInstance.reloadData(); }
+            $scope.operationInProgress = false;
         };
-
-        $scope.clearMessages = function(){
-            $scope.addPartialFailure = null;
-            $scope.removeFailure = null;
-            $scope.addWarning = null;
-            $scope.success = null;
+        $scope.updateProgressBar = function(text) {
+            /* Updates progress bar message with either `text` for a specific
+             message or the progress of the add phase of the addPeopleToCourse
+             flow if `text` is missing.
```

Comment:
super nitpicky, but maybe explain the logic behind when we use "Adding X" and when we switch to "Finishing"?